src/Parser.zig

```diff
@@ -2076,6 +2087,40 @@ const Enumerator = struct {
         e.res = res;
         // TODO adjust res type to try to fit with the previous type
     }
+
+    fn tagSpec(e: *const Enumerator, p: *Parser, is_packed: bool, tok: TokenIndex) !Type.Specifier {
+        const char_width = (Type{ .specifier = .schar }).sizeof(p.comp).? * 8;
+        const short_width = (Type{ .specifier = .short }).sizeof(p.comp).? * 8;
+        const int_width = (Type{ .specifier = .int }).sizeof(p.comp).? * 8;
+        if (e.num_negative_bits > 0) {
```

Comment:
I'm not sure if it matters currently but this should also probably include `or p.comp.langopts.emulate == .msvc or (p.comp.langopts.emulate == .clang and p.comp.target.os.tag == .windows)`. This might not be complete either but I'm assuming that mingw at least behaves like GCC.

Reply:
That's a good point; I'm not sure how clang currently handles that. This is what I currently get with `zig translate-c`:

```c
enum E { A };
```

**zig translate-c -target x86_64-windows-msvc**
```zig
pub const enum_E = c_int;
```

**zig translate-c -target x86_64-windows-gnu**
```zig
pub const enum_E = c_uint;
```

I can dig into the clang code a little bit and see if I can figure out how it decides that msvc should use an `int` even if there are no negative values. And if it's not obvious, we can work around it for now in that if statement?

Reply:
Ok so it seems like MSVC strictly follows the standard, so enums declared using the standard syntax are *always* `int`, even if enumerator constants are too large or `__attribute__((packed))` is set (can be verified using `zig translate-c`).

However MSVC (and clang) also supports a custom syntax for specifying the underlying int type of an enum:
```c
enum E : unsigned char { A };
```

**zig translate-c -target x86_64-linux-gnu test.c**
```zig
pub const enum_E = u8;
```

**zig translate-c -target x86_64-windows-msvc**
```zig
pub const enum_E = u8;
```

Reply:
> Ok so it seems like MSVC strictly follows the standard, so enums declared using the standard syntax are _always_ `int`, even if enumerator constants are too large or `__attribute__((packed))` is set (can be verified using `zig translate-c`).

Huh, didn't expect that.
 
> However MSVC (and clang) also supports a custom syntax for specifying the underlying int type of an enum:

That's a C++ thing, but based on quick testing on https://c.godbolt.org clang also supports it for C while MSVC and GCC only allow it for C++.